### PR TITLE
Only insert `""` to head of `sys.path` if a venv PEX runs in interpreter mode

### PR DESCRIPTION
**Pull request by thejcannon**
Fixes https://github.com/pantsbuild/pex/issues/1983.

Note that this PR doesn't revert the code to use the prior code (`sys.path.extend(os.environ.get("PEX_EXTRA_SYS_PATH", "").split(os.pathsep))`) as that might still add `""` when `PEX_EXTRA_SYS_PATH` is not in the environment.

Also note that this doesn't change anything related to https://github.com/pantsbuild/pex/issues/1982 (whose outcome might be that `""` is in `sys.path`, but for other reasons and in unrelated cases).

### CODE REVIEW

File: pex/venv/pex.py [hunk 1]
```diff
@@ -440,9 +440,6 @@ def sys_executable_paths():
 
             os.environ["VIRTUAL_ENV"] = venv_dir
 
-            # A Python interpreter always inserts the CWD at the head of the sys.path.
-            sys.path.insert(0, "")
```
Comment (jsirois) [inline pex/venv/pex.py:444]: I need to think through what PEX should be aiming for, but this is certainly not right no matter how that shakes out. In the zipapp execution mode, for example, "" is inserted, but only when the PEX runs in interpreter mode:
https://github.com/pantsbuild/pex/blob/67fc7e05315fc347bcf269712814f160c6035efe/pex/pex.py#L627-L631

Here in venv execution mode you've just wiped out "" insertion in all situations; so they don't behave the same.

Comment (thejcannon) [reply pex/venv/pex.py:444]: Ah I see, thanks for the code link. I'll dive further.

File: pex/venv/pex.py [hunk 2]
```diff
@@ -489,6 +489,12 @@ def sys_executable_paths():
                 os.execv(script_path, [script_path] + sys.argv[1:])
 
             pex_interpreter = pex_overrides.get("PEX_INTERPRETER", "").lower() in ("1", "true")
+
+            if pex_interpreter:
```
Comment (jsirois) [inline pex/venv/pex.py:493]: I think this needs to actually test `entry_point == PEX_INTERPRETER_ENTRYPOINT` down further. If the venv pex was built with no entrypoint, then `{entry_point!r}` will evaluate to the `None` literal and the `entry_point` will be assigned `code:interact` which you'll have missed as things stand currently.

Comment (thejcannon) [reply pex/venv/pex.py:493]: Ah right, I forgot about the non-env-var case.

**Comment by jsirois** [comment]
CI fix here: https://github.com/pantsbuild/pex/pull/1985

**Review by jsirois — APPROVED** [review]

**Pull request merged**